Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -2583,19 +2583,22 @@ RefPtr<CSSStyleValue> Parser::parse_number_value(TokenStream<ComponentValue>& to
 
 RefPtr<CSSStyleValue> Parser::parse_number_percentage_value(TokenStream<ComponentValue>& tokens)
 {
-    auto peek_token = tokens.next_token();
-    if (peek_token.is(Token::Type::Number)) {
-        tokens.discard_a_token(); // number
-        return NumberStyleValue::create(peek_token.token().number().value());
-    }
-    if (peek_token.is(Token::Type::Percentage)) {
-        tokens.discard_a_token(); // percentage
-        return PercentageStyleValue::create(Percentage(peek_token.token().percentage()));
-    }
-    if (auto calc = parse_calculated_value(peek_token); calc && calc->resolves_to_number_percentage()) {
-        tokens.discard_a_token(); // calc
-        return calc;
-    }
+    // Parses [<percentage> | <number>] (which is equivalent to [<alpha-value>])
+    if (auto number = parse_number_value(tokens))
+        return number;
+    if (auto percentage = parse_percentage_value(tokens))
+        return percentage;
+
+    return nullptr;
+}
+
+RefPtr<CSSStyleValue> Parser::parse_number_percentage_none_value(TokenStream<ComponentValue>& tokens)
+{
+    // Parses [<percentage> | <number> | none] (which is equivalent to [<alpha-value> | none])
+    if (auto value = parse_number_percentage_value(tokens))
+        return value;
+    if (auto keyword = parse_keyword_value(tokens); keyword && keyword->as_keyword().keyword() == Keyword::None)
+        return keyword;
```

Comment:
If there's a keyword here that isn't `none`, the token has still been consumed. You need to wrap this in a `TokenStream` transaction, and then only commit it if the keyword was `none`.

Reply:
I hadn't thought of that, of course!

Reply:
Fixed